src/inference/src/ie_core.cpp

```diff
@@ -747,6 +756,9 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
                 if (disabled)
                     return;
             }
+            if (!coreConfig.flag_allow_auto_batching) {
+                return;
+            }
```

Comment:
why not check the core config at the very beginning, like before GetCPPPluginByName("BATCH")

Reply:
because `config` may contain allow_auto_batch while it's disabled globally. 
Can we add a test for such use case? That local config has higher priority over globally set allow_auto_batch

Reply:
but the current logic does not put local config higher priority over globally. even I load network with auto batch enabled, it will return at line#760 finally.

Reply:
line#760 is disable auto batch. When you set batch size, the auto batching will disable.

Reply:
> because `config` may contain allow_auto_batch while it's disabled globally. Can we add a test for such use case? That local config has higher priority over globally set allow_auto_batch
I have added testcase in commit [ca990e1](https://github.com/openvinotoolkit/openvino/pull/12553/commits/ca990e1a2a949f7e80533a5e15a3166842509b92)

Reply:
@RICKIE777 what's the expected result for below scenario?
1) ie.set_property(allow_auto_batch(false))
2) ie.load_network(device, allow_auto_batch(true));

Reply:
Update. When we set "allow_auto_batching" in load_network, we will ignore the config value in global.